app/models.js

```diff
@@ -178,6 +178,45 @@ class Pod extends K8sModel {
   static get endpoint() {
     return 'pods';
   }
+
+  get display_status() {
+    let containerStatus = this.data.status.containerStatuses;
+    let display_state = 'running';
+    let reason = undefined;
+
+    if (containerStatus) {
+      for (let i = containerStatus.length - 1; i >= 0; i--) {
```

Comment:
As far as I can tell you're not using the `i` index so you could want to replace this loop with:

```
for (let status of containerStatus) {
  let state = status.state;
  // ...
```

This has the benefit to be more readable.

... hang on a second, this is looping the statuses in the reverse order, why? (genuine question)

Reply:
Are they in some kind of chronological order and you want to only the status in the first element of the array?

Also, as `display_rate` and `reason` are override by the last loop iteration, couldn't you just set these based on the first element of the array?

Reply:
I may miss something fundamental here but potentially you can get rid of the look:

```javascript
  get display_status() {
    let containerStatus = this.data.status.containerStatuses;
    let display_state = 'unknown';
    let reason = undefined;

    if (containerStatus) {
      const state = containerStatus[0].state;

      if (state.waiting) {
        display_state = 'waiting';
        reason = state.waiting.reason;
        return `Waiting: ${reason}`;
      }

      if (state.terminated) {
        display_state = 'terminated';
        reason = state.terminated.reason;
        if (!reason) {
          if (state.terminated.signal) {
            reason = `Signal:${state.terminated.signal}`;
          } else {
            reason = `ExitCode:${state.terminated.exitCode}`;
          }
        }
        return `Terminated: ${reason}`;
      }
    }

    return this.data.status.phase;
  }
```

**NOTE**: It seems to work but obviously I may miss something here.

Reply:
In fact, you probably can also get rid of `display_state` and other variables :) 

```javascript
  get display_status() {
    const containerStatus = this.data.status.containerStatuses;

    if (containerStatus) {
      const state = containerStatus[0].state;

      if (state.waiting) {
        return `Waiting: ${state.waiting.reason}`;
      }

      if (state.terminated) {
        let reason = state.terminated.reason;
        if (!reason) {
          if (state.terminated.signal) {
            reason = `Signal:${state.terminated.signal}`;
          } else {
            reason = `ExitCode:${state.terminated.exitCode}`;
          }
        }
        return `Terminated: ${reason}`;
      }
    }

    return this.data.status.phase;
  }
```

I would imagine you can squeeze this a bit further as well.

Reply:
This is a straight translation of https://github.com/kubernetes/dashboard/blob/91c54261c6a3d7f601c67a2ccfbbe79f3b6a89f9/src/app/frontend/pod/list/card_component.js#L98 - I kept the reverse iteration because I don't know why they're doing it in Go - I assume a pointless optimisation (it's usually faster to loop down to zero).
I kept the code the same to avoid introducing any bugs.